### PR TITLE
Multi-threading

### PR DESCRIPTION
**Pull request by sebdeckers**
Tried to get it working on RasPi. Needs to be more reliable and scale to low-power multi-core hardware. Went with the new Node.js worker threads. Seems to work well enough.

#### WIP

- [x] docs
- [x] CLI
- [x] systemd notify & heartbeat

#### Changes

- chore: bump deps
- docs: raspi notes
- feat: worker threads to handle the HTTP/2 session
- feat: auto reconnect
- feat: GET & POST support based on URI template syntax
- feat: multiple local listeners
- feat: multiple DoH upstreams
- feat: shortnames for popular resolvers
- feat: CLI interface redesign
- feat: HTTP error handling

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,17 @@ Options:
   -h, --help         output usage information
 ```
 
+## Deploying on a Raspberry Pi
+
+```
+sudo apt-get install git
+sudo setcap cap_net_bind_service=+ep `readlink -f \`which node\``
```
Comment (qoelet) [inline README.md]: What does this magical line do?

Comment (sebdeckers) [reply README.md]: https://serverfault.com/questions/665709/allowing-node-js-applications-to-run-on-port-80

Comment (sebdeckers) [reply README.md]: It's a cheap way to give a programme privileged port access.

Planning to use systemd/launchd mechanisms instead. They can listen on the port as PID 1 and pass the socket to our process as a FD. Haven't got that working yet.

**Comment by sebdeckers** [comment]
Sorry about the force push. Had a bad case of the gits.

**Comment by sebdeckers** [comment]
Done for now. Let's set up tests & CI in a separate PR. Will be happy to see this version roll out. Can then start working on tutorials/docs for systemd+raspi setups.

**Comment by sebdeckers** [comment]
With OOB approval from @qoelet I'm merging this PR.

**Pull request merged**